### ISSUE TITLE
[naming-convention] only enforcing rules for class/interface definition, not utilisation

### ISSUE THREAD

**Issue by Zamralik** [typescript-eslint/typescript-eslint#2759]
- [✓] I have tried restarting my IDE and the issue persists.
- [✓] I have updated to the latest version of the packages.
- [✓] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
    "rules": {
        "@typescript-eslint/naming-convention": [
            "error",
            {
                "selector": "memberLike",
                "format": ["strictCamelCase"]
            }
        ]
    }
}
```

```TS
const SERVER: Server = new Server({
    IncomingMessage: CustomIncomingMessage,
    ServerResponse: CustomServerResponse,
});
```

**Expected Result**

No error reported from the linting as it's compliant to the interface ServerOptions.

**Actual Result**

Both properties are causing an error while complying to the interface ServerOptions.

**Additional Info**

It would be better if the check could be narrowed to definitions of classes and interfaces rather than when using them.
It would avoid issues when using external components that don't follows the naming rules.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.6.1` |
| `@typescript-eslint/parser`        | `4.6.1` |
| `@types/node`                     | `14.14.6` |
| `TypeScript`                       | `4.0.5` |
| `ESLint`                           | `7.12.1` |
| `node`                             | `15.0.1` |

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/2244
